### DISCUSSION TITLE
Memory Leak issue on Docker Engine 24.0.6

### DISCUSSION THREAD

**Discussion by sachinkonde30** [moby/moby#47320] [Q&A]
Docker related issue: Here we are running our Linux based application inside docker container & when we start the process it holds 1.4GB of memory we haven’t explicitly allocated the memory neither it’s memory leakage from app side .
On normal VMs this process doesn’t hold any memory if we compared with Docket instance .
This causes Out Of Memory for docker container

The state of Process is idle still its using the High RAM. Later when we start the processes it runs in Out of memory.

Expected behavior - the reported process sp_cop should not hold 1.4G of memory.

We are using below docker version
[root@pslrhel7docker01 .ssh]# docker info
Client: Docker Engine - Community
Version: 24.0.6
Context: default
Debug Mode: false
Plugins:
buildx: Docker Buildx (Docker Inc.)
Version: v0.12.1
Path: /usr/libexec/docker/cli-plugins/docker-buildx
compose: Docker Compose (Docker Inc.)
Version: v2.24.2
Path: /usr/libexec/docker/cli-plugins/docker-compose

Server:
Containers: 1
Running: 1
Paused: 0
Stopped: 0
Images: 1
Server Version: 24.0.6
Storage Driver: overlay2
Backing Filesystem: xfs
Supports d_type: true
Using metacopy: false
Native Overlay Diff: true
userxattr: false
Logging Driver: json-file
Cgroup Driver: cgroupfs
Cgroup Version: 1
Plugins:
Volume: local
Network: bridge host ipvlan macvlan null overlay
Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
Swarm: inactive
Runtimes: runc io.containerd.runc.v2
Default Runtime: runc
Init Binary: docker-init
containerd version: a1496014c916f9e62104b33d1bb5bd03b0858e59
runc version: v1.1.11-0-g4bccb38
init version: de40ad0
Security Options:
seccomp
Profile: builtin
Kernel Version: 3.10.0-1160.el7.x86_64
Operating System: Red Hat Enterprise Linux
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 15.51GiB
Name: pslrhel7docker01
ID: 7c1b0533-f16f-4ac1-b981-39ed9b9c00e8
Docker Root Dir: /var/lib/docker
Debug Mode: false
Experimental: false

**Comment by sachinkonde30** [comment]
Below is the one with the memory issue on docker. 
![image (1)](https://github.com/moby/moby/assets/62471221/ac986cd7-fcdc-4c3e-9343-d4009fda1c0c)


The Same process in idle state is not using the memory on Local VM without docker.
<img width="708" alt="image" src="https://github.com/moby/moby/assets/62471221/cfa6241b-c6f0-442c-b2b4-c0202b8e0868">

**Comment by neersighted** [comment]
If I had to guess, this is caused by too-high ulimits and was solved in https://github.com/moby/moby/pull/45534 and https://github.com/containerd/containerd/pull/8924. Many otherwise well-written programs can allocate a lot of memory and iterate over $TOO_MANY file descriptors when the soft FD limit is set too high.
